src/main/java/controller/UserController.java

```diff
@@ -4,57 +4,66 @@
 import model.User;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import request.HttpRequest;
 import util.RequestParser;
 
-import java.util.*;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
 import java.util.stream.Collectors;
 
 public class UserController {
 
-        private Logger log = LoggerFactory.getLogger(getClass());
-        // 인스턴스 변수
-        private static UserController instance;
+    private Logger log = LoggerFactory.getLogger(getClass());
+    // 인스턴스 변수
+    private static UserController instance;
 
-        // private 생성자
-        private UserController() {
+    // private 생성자
+    private UserController() {
 
-        }
+    }
 
-        // 인스턴스 반환 메소드
-        public static UserController getInstance() {
-            if (instance == null) {
-                synchronized (UserController.class) {
-                    if (instance == null) {
-                        instance = new UserController();
-                    }
+    // 인스턴스 반환 메소드
+    public static UserController getInstance() {
+        if (instance == null) {
+            synchronized (UserController.class) {
+                if (instance == null) {
+                    instance = new UserController();
                 }
             }
-            return instance;
         }
+        return instance;
+    }
 
-        public String mapToFunctions(String httpMethod, String resourceUrl) {
+    public String mapToFunctions(HttpRequest httpRequest) {
 
-            String data = null;
+        String data = null;
+        String httpMethod = httpRequest.getValueByNameInRequestLine("httpMethod");
+        String resourceUrl = httpRequest.getValueByNameInRequestLine("resourceUrl");
 
-            if (resourceUrl.contains("?")) {
-                String[] parsedData = RequestParser.parseQueryParameter(resourceUrl);
-                resourceUrl = parsedData[0];
-                data = parsedData[1];
-                log.debug("resourceUrl = [{}], data = [{}]", resourceUrl, data);
-            }
+        if (resourceUrl.contains("?")) {
```

Comment:
```suggestion
        if (httpRequest.hasParameter()) {
```

HttpRequest 객체가 실제 HTTP 요청에 맞춰 멤버와 메소드를 가지면 좀 더 깔끔하게 코딩이 될 것 같아요.